README.md

```diff
@@ -390,6 +390,31 @@ sig-bytes = 128(OCTET)
 | `es512-varsig-header`  | `%x1202`    | `%x8224`        | P-521 [multicodec] prefix    |
 | `es512-hash-algorithm` | `%x13`      | `%x13`          | SHA2-512 [multicodec] prefix |
 
+### 5.3.4 Example: Webauthn
+
+``` abnf
+webauthn-varsig = webauthn-varsig-header client-data-length client-data-json authenticator-data-length authenticator-data signature
+
+webauthn-varsig-header = TODO
+client-data-length = 1*unsigned-varint
+client-data-json = *OCTET
+authenticator-data-length = 1*unsigned-varint
+authenticator-data = *OCTET
+signature = varsig-body
+```
+
+The Webauthn varsig header notifies the consumer that the signature is generated via webauthn. Verification must therefor rely on the [`client-data-json`][Webauthn Client Data JSON] JSON object and the [`authenticator-data`][Webauthn Authenticator Data] bytestring. The `client-data-json` and `authenticator-data` byte strings must conform to the formats specified by the Webauthn spec. To enable the inclusion of the varsig in a byte stream, they each require a varint-encoded length prefix (`client-data-length` and `authenticator-data-length` respectively).
+
+The signature itself is encoded as a varsig-body, containing hash and encoding info in the `signature` field. For example, a webauthn authenticator using `P-256` keys would result in a `signature` field which is a varsig-body conforming to the [`ES256` varsig body defined in section 5.3.1](#5.3.1-example:-es256). The signed payload is included in the signed data as the `challenge` field of the `client-data-json` as a [Multihash][Multihash]. This multihash MUST be made using the result of encoding the payload according to the parameters defined by the `signature` varsig body (for example, a DAG-CBOR encoded payload must have `signature.encoding-info` set to match). The `signature` field MUST be a signature type supported by the WebAuthn specification.
```

Comment:
Why use `challenge = multihash` and not `challenge = cid`? This way we would not need the `signature.encoding-info` field. (Afaik, the challenge can be longer than 32 bytes).

Reply:
so that the other defined varsigs could be entirely reused instead of defining cut-down versions specific to webauthn. there will also be a tiny saving on length as the encoding won't be part of a base64 string, but that wasn't my main reason

Reply:
Ok, it would be helpful if you provided an example here to make the discussion easier.

Not sure what you mean by "base64 string"

Reply:
will add an example 👍. For the base64 string, the [`client-data-json`](https://www.w3.org/TR/webauthn-2/#dictionary-client-data)'s challenge field which gets signed over is always a base64 encoding of the [challenge which is passed in (as a byte array)](https://www.w3.org/TR/webauthn-2/#dictionary-assertion-options). Because the `client-data-json` is passed back as a byte array of what was signed, we can't repack it to make anything in there smaller.

Reply:
Btw, it might make sense to sign over the CID regardless in light of #12

Reply:
the CID doesnt capture the signature algorithm though, and indeed when it comes to webauthn there is no way to embed the signature algorithm in the challenge as the authenticator selection is up to the user via the UI. I can though see how it might be useful to capture the payload encoding in the signed data